FreeRTOS_DHCP.c

```diff
@@ -570,10 +627,27 @@
 
                         if( xDHCPSocket != NULL )
                         {
+                            uint32_t ulID;
+
+                            if( xApplicationGetRandomNumber( &( ulID ) ) != pdFALSE )
+                            {
+                                EP_DHCPData.ulTransactionId = ulID;
```

Comment:
What will be the transaction ID in case `xApplicationGetRandomNumber` fails? We will use the old transaction ID?

Reply:
I had doubts here. The lease is expired. In the earlier versions, we would continue with the same transaction ID. But is is better to use a new ID. And what if the randomiser is broken?
It is very unlikely that it a randomiser breaks, once it works it will continue to work.

> We will use the old transaction ID?

So yes, extend the lease contract using the same transaction ID.

Reply:
Makes sense. Thanks, @htibosch.